### PR TITLE
[newgrp] full implementation

### PR DESCRIPTION
**Pull request by Wandalen**

### CODE REVIEW

**Comment by Wandalen** [comment]
@jgarzik ready for review

File: users/Cargo.toml [hunk 1]
```diff
@@ -14,12 +14,14 @@ gettext-rs.workspace = true
 libc.workspace = true
 chrono.workspace = true
 syslog = "6.1"
+libcrypt-rs = "0.1.2"
```
Comment (jgarzik) [inline users/Cargo.toml]: Always delete the micro version from dependencies.

File: users/newgrp.rs [hunk 2]
```diff
@@ -0,0 +1,849 @@
...
+#[command(author, version, about, long_about)]
+struct Args {
+    /// Change the environment to what would be expected if the user actually logged in again (letter `l`).
+    #[arg(short = 'l')]
+    login: bool,
+
+    /// Specifies the group ID or group name. This is a positional argument that must be provided.
+    #[arg(value_name = "GROUP", required = true)]
+    group: String,
+}
```
Comment (jgarzik) [inline users/newgrp.rs:61]: See datetime/cal.rs for an example of the new style of `struct Args` with translation support added.

Follow cal.rs style of struct Args.

File: users/newgrp.rs [hunk 3]
```diff
@@ -0,0 +1,849 @@
...
+        if !login_ptr.is_null() {
+            CStr::from_ptr(login_ptr)
+                .to_str()
+                .ok()
+                .map(|s| s.to_string())
+        } else {
+            None
+        }
+    }
+}
```
Comment (jgarzik) [inline users/newgrp.rs]: use plib::curuser::login_name() ?

File: users/newgrp.rs [hunk 4]
```diff
@@ -0,0 +1,849 @@
...
+                    s[5..].to_string()
+                } else {
+                    s.to_string()
+                }
+            })
+        } else {
+            None
+        }
+    }
+}
```
Comment (jgarzik) [inline users/newgrp.rs]: use plib::curuser::tty?

File: users/newgrp.rs [hunk 5]
```diff
@@ -0,0 +1,849 @@
...
+        }
+
+        // Exit with a non-zero status code
+        std::process::exit(1);
+    });
+
+    setlocale(LocaleCategory::LcAll, "");
+    textdomain(PROJECT_NAME)?;
+    bind_textdomain_codeset(PROJECT_NAME, "UTF-8")?;
+    let mut exit_code = 0;
```
Comment (jgarzik) [inline users/newgrp.rs]: new style:  put translation setup *before* Args parsing.

**Review by jgarzik — CHANGES_REQUESTED** [review]
small cleanups needed.  no major problems.

**Comment by Wandalen** [comment]
@jgarzik also ready

**Review by jgarzik — APPROVED** [review]

**Pull request merged**